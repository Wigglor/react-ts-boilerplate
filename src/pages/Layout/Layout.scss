// nav {
//   background-color: cadetblue;
//   ul {
//     margin: 0;
//   }
// }

// main {
.sidebar {
  background-color: cadetblue;
  color: #333;
  height: 100%;
  width: 10%;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  ul {
    list-style: none;
    padding: 0;
    li {
      div {
        svg {
          text-align: center;
          vertical-align: middle;
        }
        span {
          // text-align: center;
          vertical-align: middle;
        }

        // p {
        //   display: inline-block;
        //   margin: 0;
        //   padding-left: 0.5rem;
        // }
      }
    }
  }
}
// }
