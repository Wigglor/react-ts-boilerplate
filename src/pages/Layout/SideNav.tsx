export interface IAppProps {
  name: string;
}

export function SideNav(props: IAppProps) {
  console.log(props);
  return <div></div>;
}
