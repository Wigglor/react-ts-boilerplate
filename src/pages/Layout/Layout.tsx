import { ReactElement } from "react";
import { CiExport } from "react-icons/ci";
import { GoTasklist } from "react-icons/go";
import { MdManageAccounts } from "react-icons/md";
import { SlGraph } from "react-icons/sl";
import { Link, Outlet } from "react-router-dom";
import "./Layout.scss";

const Layout = (): ReactElement => {
  return (
    <>
      {/* <main> */}
      {/* <nav>
          <ul>
            <li>
              <Link to="/">Home</Link>
            </li>
            <li>
              <Link to="/login">Login</Link>
            </li>
            <li>
              <Link to="/dashboard">Dashboard</Link>
            </li>
          </ul>
        </nav> */}
      <div className="sidebar">
        {/* <div> */}
        <div>Logo</div>
        <ul>
          <li>
            <div>
              <Link to="/tasks">
                <GoTasklist />
                <span>Tasks</span>
              </Link>
            </div>
          </li>
          <li>
            <div>
              <Link to="/visualization">
                <SlGraph />
                <span>Visualization</span>
              </Link>
            </div>
          </li>
          <li>
            <div>
              <Link to="/exports">
                <CiExport />
                <span>Exports</span>
              </Link>
            </div>
          </li>
          <li>
            <div>
              <Link to="/account">
                <MdManageAccounts />
                <span>Account</span>
              </Link>
            </div>
          </li>
        </ul>
        {/* </div> */}
      </div>
      {/* </main> */}

      <Outlet />
    </>
  );
};

export default Layout;
