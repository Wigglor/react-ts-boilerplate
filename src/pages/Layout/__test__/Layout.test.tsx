import { /*Enzyme,*/ shallow } from "enzyme";
import toJson from "enzyme-to-json";
// import Adapter from "enzyme-adapter-react-16";
import React from "react";
import Layout from "../Layout";

// Enzyme.configure({ adapter: new Adapter() });

describe("Render Layout component", () => {
  it("renders without crashing", () => {
    shallow(<Layout />);
  });

  it("check for jsx elements", () => {
    const wrapper = shallow(<Layout />);
    const sidebar = <div className="sidebar"></div>;
    // const value = wrapper.find("h2").text();
    expect(wrapper.contains("Visualization")).toEqual(true);
    // expect(value).toEqual("hej");
  });

  it("renders correctly", () => {
    const tree = shallow(<Layout />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
