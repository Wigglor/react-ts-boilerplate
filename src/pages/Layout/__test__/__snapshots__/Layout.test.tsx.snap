// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Layout component renders correctly 1`] = `
<Fragment>
  <div
    className="sidebar"
  >
    <div>
      Logo
    </div>
    <ul>
      <li>
        <div>
          <Link
            to="/tasks"
          >
            <GoTasklist />
            <span>
              Tasks
            </span>
          </Link>
        </div>
      </li>
      <li>
        <div>
          <Link
            to="/visualization"
          >
            <SlGraph />
            <span>
              Visualization
            </span>
          </Link>
        </div>
      </li>
      <li>
        <div>
          <Link
            to="/exports"
          >
            <CiExport />
            <span>
              Exports
            </span>
          </Link>
        </div>
      </li>
      <li>
        <div>
          <Link
            to="/account"
          >
            <MdManageAccounts />
            <span>
              Account
            </span>
          </Link>
        </div>
      </li>
    </ul>
  </div>
  <Outlet />
</Fragment>
`;
